packages/devextreme/testing/tests/DevExpress.ui.widgets/toolbar.menu.tests.js

```diff
@@ -595,6 +599,85 @@ QUnit.module('widget sizing render', moduleConfig, () => {
+            if(shouldSkipOnMobile(assert)) {
+                return;
+            }
+
+            this.instance.option('closeOnClick', true);
+            this.instance.option('opened', false);
+            this.overflowMenu.$button().focusin();
+
+            this.keyboard.keyDown('enter');
+            const popup = this.overflowMenu.popup();
+            assert.equal(popup.option('visible'), true, 'popup is visible');
+
+            this.instance.option(
+                'items', [
+                    {
+                        template: () => $('<div>').dxCheckBox({ value: false }),
+                        widget: 'dxCheckBox',
+                    },
+                    {
+                        template: () => $('<div>').dxButton({ text: 'Button' })
+                    },
+                ]
+            );
+
+            const checkBox = this.overflowMenu.$popupContent().find(`.${CHECK_BOX_CLASS}`).first();
+            const button = this.overflowMenu.$popupContent().find(`.${BUTTON_CLASS}`).first();
+
+            this.instance.option({ focusStateEnabled: true });
+            this.overflowMenu.list().option({ focusStateEnabled: true });
+
+            // checkBox.focusin();
+            keyboardMock(checkBox)
+                .keyDown('space');
+            assert.equal(popup.option('visible'), true, 'space on dxCheckbox does not hide popup');
+
+            // button.focusin();
+            keyboardMock(button)
+                .keyDown('space');
+            assert.equal(popup.option('visible'), false, 'space on dxButton item hides popup');
+        });
+
+        QUnit.test('enter on list item hides drop-down list (T1301705)', function(assert) {
+            if(shouldSkipOnMobile(assert)) {
+                return;
+            }
+
+            this.instance.option('closeOnClick', true);
+            this.instance.option('opened', false);
+            this.overflowMenu.$button().focusin();
+
+            this.keyboard.keyDown('enter');
+            const popup = this.overflowMenu.popup();
+            assert.equal(popup.option('visible'), true, 'popup is visible');
+
+            this.instance.option(
+                'items', [
+                    {
+                        template: () => $('<div>').dxCheckBox({ value: false }),
+                        widget: 'dxCheckBox',
+                    },
+                    {
+                        template: () => $('<div>').dxButton({ text: 'Button' })
+                    },
+                ]
+            );
+
+            const checkBox = this.overflowMenu.$popupContent().find(`.${CHECK_BOX_CLASS}`).first();
+            const button = this.overflowMenu.$popupContent().find(`.${BUTTON_CLASS}`).first();
+
+            keyboardMock(checkBox)
+                .keyDown('enter');
+            assert.equal(popup.option('visible'), true, 'enter on dxCheckbox does not hide popup');
+
+            keyboardMock(button)
+                .keyDown('enter');
```

Comment:
This test is completely the same as the test above except the key we press? Then we can reduce copypaste if we iterate with forEach. It'll be smth like
```javascript
['space', 'enter'].forEach(key => {
            QUnit.test(`popup should be closed after ${key} press on a list item nested button`, function(assert) {
```

Reply:
are you sure that writing non explicit test cases is a good idea?
they are the same now, but it may change in the future.
for example space and enter keys have a different effect on checkbox itself.